security/backends/elasticsearch/models.py

```diff
@@ -0,0 +1,566 @@
+import json
+
+from django.db import router
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.auth import get_user_model
+from django.utils.functional import cached_property
+from django.core.serializers.json import DjangoJSONEncoder
+
+from elasticsearch_dsl import Q
+from elasticsearch_dsl import Document, Date, Integer, Keyword, Text, Object, Boolean, Float, CustomField
+
+from security.config import settings
+from security.enums import (
+    RequestLogState, CeleryTaskInvocationLogState, CeleryTaskRunLogState, CommandState, LoggerName
+)
+from security.backends.signals import (
+    input_request_started, input_request_finished, input_request_error,
+    output_request_started, output_request_finished, output_request_error,
+    command_started, command_output_updated, command_finished, command_error,
+    celery_task_invocation_started, celery_task_invocation_triggered, celery_task_invocation_ignored,
+    celery_task_invocation_timeout, celery_task_invocation_expired,
+    celery_task_run_started, celery_task_run_succeeded, celery_task_run_failed, celery_task_run_retried,
+    celery_task_run_output_updated, get_backend_receiver
+)
+
+from .app import SecurityElasticsearchBackend
+
+
+receiver = get_backend_receiver(SecurityElasticsearchBackend.backend_name)
+
+
+class JSONTextField(CustomField):
```

Comment:
special field for elasticsearch converts data to json bud not back (it is not necessary for log and it is much faster)